### PR TITLE
Keychron q9 plus

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
Added Keychron Q9 Plus and request for a review, Thanks!

## Description

- Added Q9 Pus ansi version with EC11 knob support.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q9_plus/ansi_encoder/keymaps/via/rules.mk [hunk 1]
```diff
@@ -0,0 +1,4 @@
+VIA_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
+
+OPT_DEFS += -DDYNAMIC_KEYMAP_LAYER_COUNT=5
```
[the remark this answers is not in the document]
Comment (lalalademaxiya1) [reply keyboards/keychron/q9_plus/ansi_encoder/keymaps/via/rules.mk:4]: Hello, could you please provide a format as a reference? I have tried several ways, but during compilation, none of them could correctly parse ‘dynamic_keymap.layer_count’.

[the remark this answers is not in the document]
Comment (adophoxia) [reply keyboards/keychron/q9_plus/ansi_encoder/keymaps/via/rules.mk:4]:
```json
"dynamic_keymap": {
        "layer_count": 5
},
```

[the remark this answers is not in the document]
Comment (lalalademaxiya1) [reply keyboards/keychron/q9_plus/ansi_encoder/keymaps/via/rules.mk:4]: Thanks! it works well.

**Review by drashna — APPROVED** [review]

File: keyboards/keychron/q9_plus/ansi_encoder/rules.mk [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Build Options
+#   change yes to no to disable
+#
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (fauxpark) [inline keyboards/keychron/q9_plus/ansi_encoder/rules.mk:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q9_plus/readme.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Keychron Q9 Plus
+
+![Keychron Q9 Plus](https://i.imgur.com/Yp9K7Gs.jpg)
```
Comment (drashna) [inline keyboards/keychron/q9_plus/readme.md:3]:
```suggestion
![Keychron Q9 Plus](https://i.imgur.com/Yp9K7Gsh.jpg)
```

File: keyboards/keychron/q9_plus/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+* Keyboard Maintainer: [Keychron](https://github.com/keychron)
+* Hardware Supported: Keychron Q9 Plus
+* Hardware Availability:[Keychron Q9 QMK Custom Mechanical Keyboard](https://www.keychron.com/products/keychron-q9-qmk-custom-mechanical-keyboard)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make keychron/q9_plus/ansi_encoder:default
+
+Flashing example for this keyboard ([after setting up the bootloadHID flashing environment](https://docs.qmk.fm/#/flashing_bootloadhid))
```
Comment (drashna) [inline keyboards/keychron/q9_plus/readme.md:15]: Uhhh, I don't think this is correct.  Bootloader is configured for stm32-dfu, is it not?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q9_plus/readme.md (hunk 4 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q9_plus/readme.md]: Yes, it is. the no need information has been deleted now, thanks!

**Review by drashna — APPROVED** [review]

**Comment by keychron-dev** [comment]
@fauxpark will you be able to check this recently? thanks

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**